src/Microsoft.OData.Edm/Schema/Interfaces/IEdmFullNamedElement.cs

```diff
@@ -0,0 +1,19 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="IEdmFullNamedElement.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Common base interface for all named EDM elements.
+    /// </summary>
+    public interface IEdmFullNamedElement : IEdmNamedElement
+    {
+        /// <summary>
+        /// Gets the full name of this element.
+        /// </summary>
+        string FullName { get; }

```

Comment:
This might actually be called QualifiedName. Please see my other comment regarding whether we can just use the `qualifiedName` string.

Reply:
It was called FullName before, however, instead of caching it previously, it was calculating it over and over again. Now I am simply caching it.